src/commands.rs

```diff
@@ -221,82 +290,75 @@ impl Command {
         cmd
     }
 
-    fn build_sudo_cmd(&self) -> Exec {
-        let mut cmd = Exec::cmd("sudo");
+    /// Creates a Exec that contains elevates the program with proper environment handling.
+    ///
+    /// Panics: If called when `self.elevate` is `None`
+    fn build_sudo_cmd(&self) -> Result<Exec> {
+        let elevation_program = self
+            .elevate
+            .as_ref()
+            .ok_or_else(|| eyre::eyre!("Command not found for elevation"))?
+            .resolve()
+            .context("Failed to resolve elevation program")?;
 
-        // Collect variables to preserve for sudo
-        let mut preserve_vars = Vec::new();
-        let mut explicit_env_vars = HashMap::new();
+        let mut cmd = Exec::cmd(&elevation_program);
 
-        for (key, action) in &self.env_vars {
-            match action {
-                EnvAction::Set(value) => {
-                    explicit_env_vars.insert(key.clone(), value.clone());
-                }
-                EnvAction::Preserve => {
-                    preserve_vars.push(key.as_str());
-                }
-                EnvAction::Remove => {
-                    // Explicitly don't add to preserve_vars
-                }
-            }
-        }
-
-        // Platform-agnostic handling for preserve-env
-        if !preserve_vars.is_empty() {
-            // NH_SUDO_PRESERVE_ENV: set to "0" to disable --preserve-env, "1" to force, unset defaults to force
-            let preserve_env_override = std::env::var("NH_SUDO_PRESERVE_ENV").ok();
-            match preserve_env_override.as_deref() {
-                Some("0") => {
-                    cmd = cmd.arg("--set-home");
-                }
-                Some("1") | None => {
-                    cmd = cmd.args(&[
-                        "--set-home",
-                        &format!("--preserve-env={}", preserve_vars.join(",")),
-                    ]);
-                }
-                _ => {
-                    cmd = cmd.args(&[
-                        "--set-home",
-                        &format!("--preserve-env={}", preserve_vars.join(",")),
-                    ]);
-                }
+        // Use NH_SUDO_ASKPASS program for sudo if present
+        let program_name = elevation_program
+            .file_name()
+            .and_then(|name| name.to_str())
+            .unwrap_or("");
```

Comment:
```suggestion
let program_name = elevation_program
    .file_name()
    .and_then(|name| name.to_str())
    .ok_or_else(|| eyre!("Failed to determine elevation program name"))?;
```

Silent errors are not very helpful for the user. Let's propagate the actual error.